### PR TITLE
Part 2

### PR DESCRIPTION
**Pull request by alowu**

### CODE REVIEW

File: src/main/java/com/epam/task/third/parameters/SphereParameters.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        return area;
+    }
+
+    public double getVolume() {
+        return volume;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/parameters/SphereParameters.java]: format:
if () {
}

File: src/main/java/com/epam/task/third/repository/Repository.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.epam.task.third.repository;
+
+import com.epam.task.third.observe.SphereObservable;
+import com.epam.task.third.specification.Specification;
+
+import java.util.List;
+
+public interface Repository {
+    void addSphere(SphereObservable sphere);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/repository/Repository.java]: just add

File: src/main/java/com/epam/task/third/repository/SphereRepository.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.epam.task.third.observe.SphereObservable;
+import com.epam.task.third.sort.Sort;
+import com.epam.task.third.specification.Specification;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class SphereRepository implements Sort, Repository{
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/repository/SphereRepository.java]: implements Sorter

File: src/main/java/com/epam/task/third/repository/SphereRepository.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    private static SphereRepository REPOSITORY;
+    private List<SphereObservable> list = new ArrayList<>();
+
+    private SphereRepository(){};
+
+    public static SphereRepository getRepository(){
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/repository/SphereRepository.java]: could not be a singleton

File: src/main/java/com/epam/task/third/repository/SphereRepository.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    public void removeSphere(SphereObservable sphere) {
+        list.remove(sphere);
+    }
+
+    @Override
+    public void updateSphere(SphereObservable sphere) {
+        int id = sphere.getID();
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/repository/SphereRepository.java]: int id = sphere.getId();

File: src/main/java/com/epam/task/third/sort/SortByCoordinates.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.epam.task.third.entities.Dot;
+import com.epam.task.third.entities.Sphere;
+import com.epam.task.third.observe.SphereObservable;
+
+import java.util.Comparator;
+
+public class SortByCoordinates implements Comparator<SphereObservable> {
+    @Override
+    public int compare(SphereObservable s1, SphereObservable s2) {
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/sort/SortByCoordinates.java]: better names

File: src/main/java/com/epam/task/third/specification/SpecificationByArea.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package com.epam.task.third.specification;
+
+import com.epam.task.third.logic.SphereLogic;
+import com.epam.task.third.observe.SphereObservable;
+
+public class SpecificationByArea implements Specification{
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/specification/SpecificationByArea.java]: AreaSpecification

File: src/main/java/com/epam/task/third/specification/SpecificationByVolume.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+public class SpecificationByVolume implements Specification{
+    private double volume;
+    private SphereLogic calculator = new SphereLogic();
+
+    public SpecificationByVolume(double volume) {
+        this.volume = volume;
+    }
+
+    @Override
+    public Boolean specified(SphereObservable sphere) {
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/third/specification/SpecificationByVolume.java]: boolean

File: src/test/java/com/epam/task/third/observer/SphereObserverTest.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+public class SphereObserverTest {
+    private final SphereObserver OBSERVER = SphereObserver.getObserver();
+    private final SphereObservable SPHERE = new SphereObservable(new Dot(), 1, 0);
+
+    @Test
+    public void testUpdateShouldObservedChangesWhenSphereObservableElementApplied() {
+        SPHERE.addObserver(OBSERVER);
+
+        Map<Integer, SphereParameters> stage1 = OBSERVER.getParametersMap();
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/third/observer/SphereObserverTest.java]: don't use numbers in names

File: src/test/java/com/epam/task/third/observer/SphereObserverTest.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.Map;
+
+public class SphereObserverTest {
+    private final SphereObserver OBSERVER = SphereObserver.getObserver();
+    private final SphereObservable SPHERE = new SphereObservable(new Dot(), 1, 0);
+
+    @Test
+    public void testUpdateShouldObservedChangesWhenSphereObservableElementApplied() {
+        SPHERE.addObserver(OBSERVER);
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/third/observer/SphereObserverTest.java:19]: define your given when then

File: src/test/java/com/epam/task/third/repository/SphereRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    @Test
+    public void testQueryShouldReturnListSortedByCoordinateWhenOvalsApplied(){
+        SortByCoordinates byX = new SortByCoordinates();
+
+        REPOSITORY.sort(byX);
+        List<SphereObservable> result = REPOSITORY.getList();
+
+        Assert.assertEquals(EXPECTED_COORDINATE, result);
+    }
+}
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/third/repository/SphereRepositoryTest.java:85]: where other tests?
two parts should be in one PR

**Review by FilippStankevichTR — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/task/third/repository/SphereRepositoryTest.java (hunk 11 above)
Comment (alowu) [reply src/test/java/com/epam/task/third/repository/SphereRepositoryTest.java:85]: my bad. i divided it to two PR

**Pull request merged**